Tests/Client/Custom Assertions/AssertNetworkRequest.swift

```diff
@@ -9,112 +9,251 @@ import XCTest
+    ///
+    static func networkRequest(method: Endpoint.Method,
+                               path: String,
+                               headers: [String: String]?,
+                               queryParameters: [String: String]?,
+                               body: [String: Any]?,
+                               timeout: TimeInterval = 0.5,
+                               file: StaticString = #file,
+                               line: UInt = #line) {
+        AssertAsync {
+            Assert.networkRequest(method: method,
+                                  path: path,
+                                  headers: headers,
+                                  queryParameters: queryParameters,
+                                  body: body,
+                                  timeout: timeout,
+                                  file: file,
+                                  line: line)
         }
     }
-    
-    // Check method
-    if method.rawValue != request.httpMethod {
-        errorMessage += "\n  - Incorrect method: expected \"\(method.rawValue)\" got \"\(request.httpMethod ?? "_")\""
+}
+
+extension Assert {
+    /// Synchronously waits for a network request that matches its properties with the given parameters.
+    ///
+    /// The function periodically checks the `RequestRecorderURLProtocol.recordedRequests` and If no request matches the
+    /// given parameters within the `timeout` period, this assertion fails with the time-out error.
+    ///
+    /// The values specified in the `headers`, `queryParameters` and `body` represents the mandatory subset of
+    /// the values the request must have. A request is valid even when it contains additional parameters than
+    /// the ones specified in these values.
+    ///
+    /// - Parameters:
+    ///   - method: The HTTP method the request.
+    ///   - path: The `path` part of the request's URL.
+    ///   - headers: The headers required for the request.
+    ///   - queryParameters: The query parameters required for the request.
+    ///   - body: The expected body of the request.
+    ///   - timeout: The maximum time this function waits for a request to match the given parameters.
+    ///
+    static func networkRequest(method: Endpoint.Method,
+                               path: String,
+                               headers: [String: String]?,
+                               queryParameters: [String: String]?,
+                               body: [String: Any]?,
+                               timeout: TimeInterval = 0.5,
+                               file: StaticString = #file,
+                               line: UInt = #line) -> Assertion {
+        
+        
+        return Assert.willBeTrue(RequestRecorderURLProtocol.recordedRequests.contains {
+            $0.matches(method, path, headers, queryParameters, body).isSuccess
+        }, message: failureMessage(method, path, headers, queryParameters, body),
+           file: file,
```

Comment:
This should address this comment about not passing the file and line: https://github.com/GetStream/stream-chat-swift/pull/364#discussion_r472044227